### ISSUE TITLE
How to initiate a request using a socks5 proxy with authentication?

### ISSUE THREAD

**Issue by ljh740** [swift-server/async-http-client#750]
ex: user:pwd@127.0.0.1:10086
type: socks5
```
Proxy.socksServer(host: "127.0.0.1", port: 10086)
```
Unable to set authentication information

**Comment by Lukasa** [comment]
Async HTTP Client doesn't currently support SOCKS5 authentication, because the upstream [SOCKS support](https://github.com/apple/swift-nio-extras/tree/main/Sources/NIOSOCKS) doesn't either. We'd need a fix upstream to add that support before we could add it here.

**Comment by ljh740** [comment]
> Async HTTP Client doesn't currently support SOCKS5 authentication, because the upstream [SOCKS support](https://github.com/apple/swift-nio-extras/tree/main/Sources/NIOSOCKS) doesn't either. We'd need a fix upstream to add that support before we could add it here.

Any plans to update it in the near future?

**Comment by Lukasa** [comment]
I don't believe we have any plans to, but you'd be welcome to file an issue or tackle it yourself! We'd happily review any changes needed.
